data/mods/gen7/random-teams.ts

```diff
@@ -807,319 +908,180 @@ export class RandomGen7Teams extends RandomGen8Teams {
 		return abilityAllowed[0].name;
 	}
 
-	getHighPriorityItem(
+	getPriorityItem(
 		ability: string,
-		types: Set<string>,
+		types: string[],
 		moves: Set<string>,
 		counter: MoveCounter,
 		teamDetails: RandomTeamsTypes.TeamDetails,
 		species: Species,
 		isLead: boolean,
-		isDoubles: boolean
+		preferredType = '',
+		role = '',
 	): string | undefined {
-		if (species.requiredItems) {
-			if (
-				species.baseSpecies === 'Arceus' &&
-				(moves.has('judgment') || !counter.get(species.types[0]) || teamDetails.zMove)
-			) {
-				// Judgment doesn't change type with Z-Crystals
-				return species.requiredItems[0];
+		// Z-Moves
+		if (role === 'Z-Move user') {
+			// Specific Z-Crystals
+			if (species.baseSpecies === 'Arceus' && species.requiredItems) return species.requiredItems[1];
+			if (species.name === 'Raichu-Alola') return 'Aloraichium Z';
+			if (species.name === 'Decidueye') return 'Decidium Z';
+			if (species.name === 'Kommo-o') return 'Kommonium Z';
+			if (species.name === 'Lunala') return 'Lunalium Z';
+			if (species.baseSpecies === 'Lycanroc') return 'Lycanium Z';
+			if (species.name === 'Marshadow') return 'Marshadium Z';
+			if (species.name === 'Mew') return 'Mewnium Z';
```

Comment:
Possibly this could be simplified by iterating over all Z-crystals and checking `itemUser`, but this way is probably more performant (albeit potentially trickier to maintain).

Reply:
Agreed, I think the list is short enough that this is more performant, especially since some Z-Move users with specific Z-crystals opt not to use them, like Tapu Koko.